modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -235,6 +235,30 @@ object Inputs {
     dir
   }
 
+  private def computeCommonRoot(validElems: Seq[Element]): os.Path = {
+    val sourceDirectories = validElems.collect {
+      case s: SourceFile => s.path / os.up
+      case d: Directory  => d.path
+    }.toList
+
+    assert(sourceDirectories.nonEmpty)
+
+    def findCommonRoot(
+      candidateRootSegments: Iterator[String],
+      directories: Seq[os.Path]
+    ): os.Path =
+      directories match {
+        case dirPath :: rest =>
+          val updatedCandidate = candidateRootSegments.zip(dirPath.segments)
+            .takeWhile(Function.tupled(_ == _))
+            .map(_._1)
+          findCommonRoot(updatedCandidate, rest)
+        case Nil => os.root / candidateRootSegments.toList
```

Comment:
Not sure `os.root` works fine on Windows, when several drives (`C:\`, `D:\`, …) are involved.

Reply:
When `candidateRootSegments.toList` is empty, we should just use a directory under home dir at the end.

Reply:
What if someone runs scala-cli in the `/tmp` directory. I wouldn't expect that `scala-cli` generate the files under home dir.

Reply:
<img width="1094" alt="Screenshot 2022-02-21 at 12 20 34" src="https://user-images.githubusercontent.com/46607934/154945908-613c4c02-1fd0-4139-947f-b8d7c684e8f3.png">
It seems, that `os.root` working fine on Windows, if there are several drives

Reply:
What I mean is that `os.root` is usually `C:\` on Windows. So if we're passed files on `D:\` for example, that will be a problem. We just shouldn't rely on `os.root` on Windows.